src/__internal__/input-icon-toggle/input-icon-toggle.test.tsx

```diff
@@ -0,0 +1,358 @@
+import React from "react";
+import { render, screen, createEvent, fireEvent } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import * as floatingUi from "@floating-ui/react-dom";
+import InputIconToggle, { InputIconToggleProps } from ".";
+
+test.each(["error", "warning", "info"])(
+  "renders only a validation icon when the validation prop is set to %s as a string and `useValidationIcon` is true",
+  (validationProp) => {
+    render(
+      <InputIconToggle
+        {...{
+          [validationProp]: "Message",
+          useValidationIcon: true,
+          inputIcon: "settings",
+        }}
+      />
+    );
+
+    const validationIcon = screen.getByTestId(`icon-${validationProp}`);
+    const icon = screen.queryByTestId("icon");
+    expect(validationIcon).toBeVisible();
+    expect(icon).not.toBeInTheDocument();
+  }
+);
+
+test.each([
+  ["left", "right"],
+  ["right", "left"],
+] as [InputIconToggleProps["align"], string][])(
+  "tooltip positioningwhen align prop is passed as %s, tooltip position should be set as %s",
```

Comment:
Typo spotted here

Reply:
**suggestion**: since we are testing our usage of `@floating-ui` here, rather than the rendered position of the tooltip, what do you think about tweaking the title to reflect this, or adjusting the test to check if the tooltip has the correct `data-placement` attribute?


```tsx
render(
    <InputIconToggle align={tooltipAlign} useValidationIcon error="error" />
);

const tooltip = screen.getByRole("tooltip");
expect(tooltip).toHaveAttribute("data-placement", tooltipPosition);
```